tests/sentry/workflow_engine/processors/test_data_condition_group.py

```diff
@@ -0,0 +1,253 @@
+    evaluate_condition_group,
+    get_data_conditions_for_group,
+    process_data_condition_group,
+)
+from sentry.workflow_engine.types import DetectorPriorityLevel
+
+
+class TestGetDataConditionsForGroup(TestCase):
+    def test_get_data_conditions_for_group(self):
+        assert get_data_conditions_for_group(0) == []
+
+    def test_get_data_conditions_for_group__exists(self):
+        data_condition_group = self.create_data_condition_group()
+        data_condition = self.create_data_condition(condition_group=data_condition_group)
+        assert get_data_conditions_for_group(data_condition_group.id) == [data_condition]
+
+
+class TestProcessDataConditionGroup(TestCase):
+    def test_process_data_condition_group(self):
+        with mock.patch(
+            "sentry.workflow_engine.processors.data_condition_group.logger"
+        ) as mock_logger:
+            assert process_data_condition_group(1, 1) == (False, [])
+            assert mock_logger.exception.call_args[0][0] == "DataConditionGroup does not exist"
+
+    def test_process_data_condition_group__exists__fails(self):
+        data_condition_group = self.create_data_condition_group()
+        self.create_data_condition(
+            condition_group=data_condition_group, condition="gt", comparison="5"
+        )
+
+        assert process_data_condition_group(data_condition_group.id, 1) == (False, [])
+
+    def test_process_data_condition_group__exists__passes(self):
+        data_condition_group = self.create_data_condition_group()
+        self.create_data_condition(
+            condition_group=data_condition_group,
+            condition="gt",
+            comparison="5",
+            condition_result=DetectorPriorityLevel.HIGH,
+        )
+        assert process_data_condition_group(data_condition_group.id, 10) == (
+            True,
+            [DetectorPriorityLevel.HIGH],
+        )
+
+
+class TestEvaluateConditionGroupTypeAny(TestCase):
+    def setUp(self):
+        self.data_condition_group = self.create_data_condition_group(
+            logic_type=DataConditionGroup.Type.ANY
+        )
+
+        self.data_condition = self.create_data_condition(
+            condition="gt",
+            comparison="5",
+            condition_result=DetectorPriorityLevel.HIGH,
+            condition_group=self.data_condition_group,
+        )
+
+        self.data_condition_two = self.create_data_condition(
+            condition="gt",
+            comparison="3",
+            condition_result=DetectorPriorityLevel.LOW,
+            condition_group=self.data_condition_group,
+        )
+
+        self.conditions = [self.data_condition, self.data_condition_two]
+
+    def test_evaluate_condition_group__passes_all(self):
+        assert evaluate_condition_group(
+            self.data_condition_group,
+            10,
+        ) == (
+            True,
+            [DetectorPriorityLevel.HIGH, DetectorPriorityLevel.LOW],
```

Comment:
Do we later on check the results against the occurrence's group's existing status to determine whether to change the status and what to change it to?

Reply:
kinda - rather than checking the occurrence (because that's async and we won't have guaranteed state), we use the `DetectorStateModel` which just tracks the latest state change for the detector. (but it's more or less the same logic / reasoning to check the occurrences status)